### ISSUE TITLE
Support for upcoming `param_attrs` feature

### ISSUE THREAD

**Issue by phansch** [rust-lang/rust-clippy#4474]
[`param_attrs` is likely to be stabilized in Rust 1.39](https://github.com/rust-lang/rust/pull/64010) 

The `param_attrs` feature is going to allow lint attributes on function, method and closure parameters:

```rust
fn foo(
    a: i32,
    #[allow(unused_variables)] b: i32,
) {}
```

We should review how this feature ties in with our existing lints. Apparently it did not cause any toolstate breakage [when it was merged](https://github.com/rust-lang/rust/pull/61856).

Specifically, we should create a list of lints that could benefit from parameter attributes and then update those lints to make use of them.
